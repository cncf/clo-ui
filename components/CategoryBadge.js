module.exports = require('../dist/cjs/components/CategoryBadge.js');
