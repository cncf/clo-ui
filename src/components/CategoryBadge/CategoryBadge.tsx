import { isNull, isUndefined } from 'lodash';
import React from 'react';

import { cutString } from '../../utils';
import styles from './CategoryBadge.module.css';

export interface ICategoryBadgeProps {
  category?: string | null;
  className?: string;
  maxLength?: number;
}

export const CategoryBadge: React.FC<ICategoryBadgeProps> = (props: ICategoryBadgeProps) => {
  if (isUndefined(props.category) || isNull(props.category)) return null;

  return (
    <div
      data-testid="category-badge"
      className={`badge text-dark lighterText rounded-0 position-relative ${styles.badge} ${props.className}`}
    >
      <div className="d-flex flex-row align-items-center text-capitalize">
        {!isUndefined(props.maxLength) ? cutString(props.category, props.maxLength) : props.category}
      </div>
    </div>
  );
};
