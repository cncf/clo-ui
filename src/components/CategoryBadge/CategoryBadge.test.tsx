import { render, screen } from '@testing-library/react';

import { CategoryBadge } from './CategoryBadge';

const defaultProps = {
  category: 'sandbox',
};

describe('CategoryBadge', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<CategoryBadge {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(<CategoryBadge {...defaultProps} />);

    expect(screen.getByText('sandbox')).toBeInTheDocument();
  });
});
