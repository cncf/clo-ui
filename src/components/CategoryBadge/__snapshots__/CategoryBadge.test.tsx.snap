// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryBadge creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="badge text-dark lighterText rounded-0 position-relative badge undefined"
    data-testid="category-badge"
  >
    <div
      class="d-flex flex-row align-items-center text-capitalize"
    >
      sandbox
    </div>
  </div>
</DocumentFragment>
`;
